src/sys/unix/tcp.rs

```diff
@@ -465,7 +465,19 @@ pub fn accept(listener: &net::TcpListener) -> io::Result<(net::TcpStream, Socket
             &mut length
         ))
         .map(|socket| unsafe { net::TcpStream::from_raw_fd(socket) })
-        .and_then(|s| syscall!(fcntl(s.as_raw_fd(), libc::F_SETFD, libc::FD_CLOEXEC)).map(|_| s))
+        .and_then(|s| {
+            let mut f = syscall!(fcntl(s.as_raw_fd(), libc::F_SETFD, libc::FD_CLOEXEC));
```

Comment:
You can use `syscall!(...)?` here and ignore the result (now stored in `f`).

Reply:
Do you mean like this? Because this actually doesn't seem to work 🤔 
```rust
let stream = {
        syscall!(accept(
            listener.as_raw_fd(),
            addr.as_mut_ptr() as *mut _,
            &mut length
        ))
        .map(|socket| unsafe { net::TcpStream::from_raw_fd(socket) })
        .and_then(|s| {
            syscall!(fcntl(s.as_raw_fd(), libc::F_SETFD, libc::FD_CLOEXEC))?;
    
            // See https://github.com/tokio-rs/mio/issues/1450
            #[cfg(all(arget_arch = "x86", target_os = "android"))]
            syscall!(fcntl(s.as_raw_fd(), libc::F_SETFL, libc::O_NONBLOCK))?;
            
            Ok(s)
        })
    }?;
```

Reply:
`arget_arch` d'oh

It works fine 🙄